src/connector/src/parser/protobuf/parser.rs

```diff
@@ -134,17 +139,20 @@ impl ProtobufParserConfig {
     fn pb_field_to_col_desc(
         field_descriptor: &FieldDescriptor,
         index: &mut i32,
+        parse_trace: &mut Vec<String>,
     ) -> Result<ColumnDesc> {
-        let field_type = protobuf_type_mapping(field_descriptor)?;
+        detect_loop_and_push(parse_trace, field_descriptor)?;
+        let field_type = protobuf_type_mapping(field_descriptor, parse_trace)?;
```

Comment:
I'm considering whether we can merge these two steps so that we don't forget to call `detect`.

Reply:
fixed